src/rdkafka_telemetry_encode.c

```diff
@@ -0,0 +1,837 @@
+                    &rkb->rkb_telemetry.rd_avg_rollover.rkb_avg_outbuf_latency,
+                    &rkb->rkb_telemetry.rd_avg_current.rkb_avg_outbuf_latency);
+                rd_avg_destroy(
+                    &rkb->rkb_telemetry.rd_avg_rollover.rkb_avg_throttle);
+                rd_avg_rollover(
+                    &rkb->rkb_telemetry.rd_avg_rollover.rkb_avg_throttle,
+                    &rkb->rkb_telemetry.rd_avg_current.rkb_avg_throttle);
+        }
+
+        int resource_attributes_count =
+            resource_attributes(rk, &resource_attributes_struct);
+        rd_kafka_dbg(rk, TELEMETRY, "PUSH", "Resource attributes count: %d",
+                     resource_attributes_count);
+        if (resource_attributes_count > 0) {
+                resource_attributes_key_values =
+                    rd_malloc(sizeof(opentelemetry_proto_common_v1_KeyValue *) *
+                              resource_attributes_count);
+                int ind;
+                for (ind = 0; ind < resource_attributes_count; ++ind) {
+                        resource_attributes_key_values[ind] = rd_calloc(
+                            1, sizeof(opentelemetry_proto_common_v1_KeyValue));
+                        resource_attributes_key_values[ind]->key.funcs.encode =
+                            &encode_string;
+                        resource_attributes_key_values[ind]->key.arg =
+                            (void *)resource_attributes_struct[ind].name;
+
+                        resource_attributes_key_values[ind]->has_value = true;
+                        resource_attributes_key_values[ind]->value.which_value =
+                            opentelemetry_proto_common_v1_AnyValue_string_value_tag;
+                        resource_attributes_key_values[ind]
+                            ->value.value.string_value.funcs.encode =
+                            &encode_string;
+                        resource_attributes_key_values[ind]
+                            ->value.value.string_value.arg =
+                            (void *)resource_attributes_struct[ind].value;
+                }
+                resource_attributes_repeated.key_values =
+                    resource_attributes_key_values;
+                resource_attributes_repeated.count = resource_attributes_count;
+                resource_metrics.has_resource      = true;
+                resource_metrics.resource.attributes.funcs.encode =
+                    &encode_key_values;
+                resource_metrics.resource.attributes.arg =
+                    &resource_attributes_repeated;
+        }
+
+        opentelemetry_proto_metrics_v1_ScopeMetrics scopeMetrics =
+            opentelemetry_proto_metrics_v1_ScopeMetrics_init_zero;
+
+        opentelemetry_proto_common_v1_InstrumentationScope
+            instrumentationScope =
+                opentelemetry_proto_common_v1_InstrumentationScope_init_zero;
+        instrumentationScope.name.funcs.encode    = &encode_string;
```

Comment:
Same for `instrumentation_scope`
```suggestion
        instrumentation_scope.name.funcs.encode    = &encode_string;
```

Reply:
Updated